active/XamlServiceProvider/XamlServiceProviderApi.md

```diff
@@ -0,0 +1,251 @@
+ 
+![MarkupExtension sample](MESample.png)
+
+## MarkupExtension.ProvideValue virtual method
+
+A markup extension overrides the ProvideValue method, and returns a value from it that the Xaml loader sets as a property value.
+
+There are two versions of ProvideValue, one that passes no parameters and one that passes a service provider. From the service provider, a markup extension implementation can retrieve the following services:
+* `IProvideValueTarget`, which provides information about the type/property the markup extension is being applied to. This aligns with WPF's [IProvideValueTarget](https://docs.microsoft.com/dotnet/api/System.Windows.Markup.IProvideValueTarget).
+* `IRootObjectProvider`, which provides a reference to the root object in the markup. This aligns with WPF's [IRootObjectProvider](https://docs.microsoft.com/dotnet/api/System.Xaml.IRootObjectProvider).
+* `IUriContext` , which provides the base URI of the markup. This aligns with WPF's [IUriContext](https://docs.microsoft.com/dotnet/api/System.Windows.Markup.IUriContext).
+
+
+
+# Examples
+
+## IProvideValueTarget
+
+The following example shows a custom markup extension whose provided value changes based on the specific property being targeted by the custom markup extension. In this case, the developer wants something that will automatically use an AcrylicBrush for the [Control.Background](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Control.Background) or [Border.Background](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Border.Background) properties, but uses a [SolidColorBrush](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Media.SolidColorBrush) for all other properties.
+
+```cs
+public class BrushSelectorExtension : MarkupExtension
+{
+    public Color Color { get; set; }
+
+    protected override object ProvideValue(IXamlServiceProvider serviceProvider)
+    {
+        Brush brushToReturn = new SolidColorBrush() { Color = Color }; ;
+
+        var provideValueTarget = serviceProvider.GetService(typeof(IProvideValueTarget))
+            as IProvideValueTarget;
+        if (provideValueTarget.TargetProperty is ProvideValueTargetProperty targetProperty)
+        {
+            if (targetProperty.Name == "Background"
+                && (targetProperty.DeclaringType == typeof(Control) 
+                    || targetProperty.DeclaringType == typeof(Border)))
+            {
+                brushToReturn = new AcrylicBrush()
+                    { TintColor = Color, TintOpacity = 0.75 };
+            }
+        }
+
+        return brushToReturn;
+    }
+}
+```
+
+```xml
+<StackPanel>
+    <Button Foreground="{local:BrushSelector Color=Blue}" 
+            Background="{local:BrushSelector Color=Gold}">
+        Go bears!
+    </Button>
+    <Rectangle x:Name="SolidColor" Fill="{local:BrushSelector Color=Green}" />
+</StackPanel>
+```
+
+
+## IRootObjectProvider
+
+> Some more realistic example?

```

Comment:
Get the `Title` from the page (if the rootobject is a `Page`)?

Reply:
Page doesn't have a Title in WinUI